services/storage/src/simcore_service_storage/datcore_dsm.py

```diff
@@ -59,9 +59,21 @@ async def list_files(
     async def get_file(self, user_id: UserID, file_id: StorageFileID) -> FileMetaData:
         raise NotImplementedError
 
-    async def create_file_upload_link(
-        self, user_id: UserID, file_id: StorageFileID, link_type: LinkType
-    ) -> AnyUrl:
+    async def create_file_upload_links(
+        self,
+        user_id: UserID,
+        file_id: StorageFileID,
+        link_type: LinkType,
+        file_size_bytes: ByteSize,
+    ) -> UploadLinks:
+        raise NotImplementedError
```

Comment:
THOUGHT: this is transformed in our ``servicelib.aiohttp`` error middleware into https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/501. I wonder whether that is the right code ...  (if so, i will implemente analogous for fastapi)